src/Public/Utilities.ps1

```diff
@@ -1486,3 +1493,119 @@ function Invoke-PodeGC {
 
     [System.GC]::Collect()
 }
+
+<#
+.SYNOPSIS
+    A function to pause execution for a specified duration with an optional progress bar.
+
+.DESCRIPTION
+    The `Start-PodeSleep` function pauses script execution for a given duration specified in seconds, milliseconds, or a TimeSpan.
+    It includes an optional progress bar that displays the elapsed time and completion percentage.
+    The progress bar can also display a custom activity name and allows grouping with a ParentId.
+
+.PARAMETER Seconds
+    Specifies the duration to pause execution in seconds. Default is 1 second.
+
+.PARAMETER Milliseconds
+    Specifies the duration to pause execution in milliseconds.
+
+.PARAMETER Duration
+    Specifies the duration to pause execution using a TimeSpan object.
+
+.PARAMETER Activity
+    Specifies the activity name displayed in the progress bar. Default is "Sleeping...".
+
+.PARAMETER ParentId
+    Optional parameter to specify the ParentId for the progress bar, enabling hierarchical grouping.
+
+.PARAMETER ShowProgress
+    Switch to enable the progress bar during the sleep duration.
+
+.OUTPUTS
+    None.
+
+.EXAMPLE
+    Start-PodeSleep -Seconds 5 -ShowProgress
+
+    Pauses execution for 5 seconds and displays a progress bar.
+
+.EXAMPLE
+    Start-PodeSleep -Milliseconds 3000 -ShowProgress -Activity "Processing Task" -ParentId 1
+
+    Pauses execution for 3000 milliseconds, showing a progress bar with the custom activity grouped under ParentId 1.
+
+.EXAMPLE
+    Start-PodeSleep -Duration (New-TimeSpan -Seconds 10) -ShowProgress -Activity "Running Script"
+
+    Pauses execution for 10 seconds using a TimeSpan object and displays a progress bar.
+
+.NOTES
+    This function is useful for scenarios where tracking the remaining wait time visually is helpful.
+#>
+function Start-PodeSleep {
```

Comment:
the reason of this function is to avoid long sleep. otherwise suspend will fails
your change doesn't solve that. I can remove the progress bar because is not used anymore

Reply:
I was thinking to replace any Start-Sleep with Start-PodeSleep inside any task and schedule scriptblock